### PR TITLE
Add block contact and encryption info to profile

### LINKED ISSUE #4043: No way to do almost anything with a contact w/o having a 1:1 chat
If i only have some group with a contact, but not a 1:1 chat, i can't do almost anything with it, e.g. see "Encryption Info" or "Block Contact" (which is possible in DC Android f.e.). I only can "Edit Name". Ok, i created a 1:1 chat with the contact, now i have the extended menu (btw, i still see "Encryption Info" only in the chatlist ctx menu, but not in the chat one). Then i tried to "Block Contact", after that the 1:1 chat was deselected and i see the message "Select a chat or create a new chat". Why deselect the chat btw? I selected the 1:1 chat back, it's still empty, nothing has changed visually and if i open the menu again, i see "Block Contact" while expecting to have "Unblock Contact". So, i can block the contact again and if i do that, the same effects as described before occur. My suggestions:
- **Add the extended menu (with "Encryption Info" etc. -- as on Android) to the contact window (where the user only can "Edit Name" now).**
- Add "Encryption Info" to the 1:1 chat menu (looks like it's just forgotten, other menu items are there). **EDIT:** See the discussion, this is not needed, e.g. DC Android doesn't have it there.
- Reconsider deselecting the 1:1 chat after blocking a contact. **EDIT:** It's ok, the chat should be hidden after blocking the contact.
- Debug why a contact with an empty 1:1 chat can be blocked multiple times w/o any effects. **EDIT:** This happens not for a contact with an empty 1:1, but for an already blocked contact. Not a case for DC Android, see below. Created a separate issue: #4046.

### PR DESCRIPTION
**Pull request by maxphilippov**
- new `useViewProfileMenu` menu with two items: Block/Unblock Contact and Encryption Info
- minor refactor of `useChatDialog`, so we don't have to have a ChatListItem to block contact or get EncryptionInfo
- resolves #4043

### CODE REVIEW

**Review by nicodh — CHANGES_REQUESTED** [review]
There are 3 options to block a contact now:
1. from contact context menu in chat list
2. from context menu in chat view
3. from settings menu in ViewProfile

Depending on the use case the behaviour is different:
1. if you use option 1 while the chat is not open, everything is fine
2. if you use option 1 while the chat is open, the chat stays open
3. if you use option 2 while the chat is open, the chat is unselected and closed
4. if you use option 3 while the chat is open, the ViewProfile stays open, the chat disappears from chat list, but the chat is still open when you close the ViewProfile dialog
5. in an open chat (remained after blocking a contact) the context menu still shows: "Block contact" instead of "unblock contact"

I suggest to always use the same behaviour as in  chat views context menu, that means after blocking a contact the current chat should be unselected if it is the 1:1 chat with the blocked user. If the user is blocked from "ViewProfile" dialog the dialog should be closed too. Otherwise we would have to add a flag to show that a user is blocked.

**Comment by maxphilippov** [comment]
> 5. in an open chat (remained after blocking a contact) the context menu still shows: "Block contact" instead of "unblock contact"
There's a separate issue #4046 for this and from what I saw looking at this it seems more like a chatlist problem so I don't want to mix it up here, unless it's a minor fix, I'll push it as a separate PR.

Thanks for other points! I'll reiterate on that (+ menu styling) and push an update soon

**Comment by nicodh** [comment]
> Thanks for other points! I'll reiterate on that (+ menu styling) and push an update soon

@maxphilippov please let me know when you updated this

**Comment by maxphilippov** [comment]
@nicodh everything is ready for review. I've also moved "Edit Name" to this new menu as we discussed with @r10s. There also rises a question if we need the same menu for groups but with only "Encryption Info" + "Edit Name" but that's for another PR probably.

**Review by nicodh — APPROVED** [review]

**Comment by r10s** [comment]
> There also rises a question if we need the same menu for groups but with only "Encryption Info" + "Edit Name" but that's for another PR probably.

for groups, editing name+image is far more important than setting a nickname, so i would not suggest to move that to the "three dot menu" without other changes (eg. being able to access "Edit" also from enlarged avatar view as on android).

still, for encryption info, the three dot menu makes sense for groups, so that this is consistent.

when we have that consistent, also in another PR, we can probably also remove  "Encryption Info" from the chatlist's context menus. 

however, that is future and shall not block merging this PR :) all in all a nice improvement, thanks a lot.

**Pull request merged**
